xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/DiagramMacros.xml

```diff
@@ -74,7 +73,7 @@
         'search': 'Diagram'
       }))
       #set ($installURL = $xwiki.getURL('XWiki.XWikiPreferences', 'admin', $queryString))
-      $services.localization.render('confluencediagram.installRequired', ['[[', "&gt;&gt;path:$installURL||target='_blank']]"])
+      $escapetool.xml($services.localization.render('confluencediagram.installRequired', ['[[', "&gt;&gt;path:$installURL||target='_blank']]"]))
```

Comment:
You're not inside HTML here, so XML escaping doesn't make much sense. You need wiki syntax escaping, but note that the translation parameter values contain wiki syntax, so you can't wrap everything in an escape. The solution is normally to:

```
#set ($message = $services.localization.render('confluencediagram.installRequired', ['linkPrefix', 'linkSuffix']))
#set ($escapedMessage = $services.rendering.escape($message, ...))
#set ($escapedMessage = $escapetedMessage.replace('linkPrefix', ...).replace('linkSuffix', ...))
```

It's not straightforward but it's needed.

Reply:
fixed